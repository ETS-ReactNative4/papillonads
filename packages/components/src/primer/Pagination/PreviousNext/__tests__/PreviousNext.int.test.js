import { renderCustom, renderDefault } from './PreviousNext.int.render'

describe('<PreviousNext />', () => {
  const onClickMockFn = jest.fn()
  // const dispatchMockFn = jest.fn()

  afterEach(() => jest.clearAllMocks())

  describe('Event', () => {
    test('must return new items without index when previous onClick()', () => {
      const mountCustomRender = global.renderMount(renderCustom(onClickMockFn))
      mountCustomRender.find('a').first().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      // expect(dispatchMockFn).toHaveBeenCalledWith('onBackwards')
      // expect(mountCustomRender.props()).toBe({})
    })

    test('must return new items without index when next onClick()', () => {
      const mountCustomRender = global.renderMount(renderCustom(onClickMockFn))
      mountCustomRender.find('a').last().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      // expect(dispatchMockFn).toHaveBeenCalledWith('onForward')
      // expect(mountCustomRender.props()).toBe({})
    })
  })

  describe('Render', () => {
    test('must match renderDefault()', () => {
      expect(global.renderToJSON(renderDefault())).toMatchSnapshot()
    })
  })
})
