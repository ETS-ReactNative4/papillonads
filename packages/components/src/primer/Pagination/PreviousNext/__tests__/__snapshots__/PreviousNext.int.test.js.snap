// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviousNext /> Render must match renderDefault() 1`] = `
<nav
  aria-label="Pagination"
  className="paginate-container"
>
  <div
    className="pagination d-flex"
  >
    <a
      aria-label="Previous"
      href="#url"
      onClick={[Function]}
      rel="previous"
    >
      &lt;
    </a>
    <span>
      Viewing 3 to 7
    </span>
    <a
      aria-label="Next"
      href="#url"
      onClick={[Function]}
      rel="next"
    >
      &gt;
    </a>
  </div>
</nav>
`;
